### ISSUE TITLE
Hello, do you solution a stable version  async/await for IL2CPP?

### ISSUE THREAD

**Issue by iamchenxin** [ufcpp/AwaitableUniRx#1]
Please forgive me such simply question. 
I am not familiar with Japanese.
Thanks

**Comment by ufcpp** [comment]
Thanks to your question. Don't mind about language. I'd like to provide documents in English, but I can't because of lack of time.

## migration to MinimumAsyncBridge

I migrated to another approach: https://github.com/OrangeCube/MinimumAsyncBridge

The MinimumAsyncBridge library is a minimal implementation of Task, TaskCompletionSource, TaskAwaiter, and so on, which can be used in Unity game engine.

## stability

 I use this library in medium-scale projects at work with a little trouble. The projects can be compiled in IL2CPP on some versions of Unity, and can't on other versions. So far, the compilation succeeds on Unity 5.1.4, but fails on Unity 5.2.2. I've provided feedback on this issue to Unity, and Unity has answered to fix it in the near future. Once compiled, the result of the projects works well without any runtime exception.

## awaiter for UniRx

If you want use it for UniRx, There is an awaiter implementation for UniRx in the repository: https://github.com/OrangeCube/MinimumAsyncBridge/tree/master/src/UniRx.AsyncBridge

## limitation

This library contains classes which are required for async/await. However, the C# 5.0 (or later) compiler is also required. Thus:
- you can use async/await with .NET 3.5 and this library with the C# 5.0 compiler.
- if you create a DLL with the C# 5.0 compiler and copy it to a Unity project, you can use async/await in the DLL.
- but, you can still not use async/await with the compiler bundled with Unity.

## license problem

The implementation is based on code in https://github.com/Microsoft/referencesource, I use https://github.com/tunnelvisionlabs/dotnet-threading/pull/90 as a reference, and the MinimumAsyncBridge has the same problem as the dotnet-threading.

In short, Microsoft is working hard to make .NET Framework open-source (MIT-license), but it is in progress and the `System.Threading.Tasks` is not done yet. the MinimumAsyncBridge (and also the dotnet-threading) depends on source codes whose license is ambiguous.

**Comment by iamchenxin** [comment]
Thanks again.
I will try the MinimumAsyncBridge.
It seems all the thing will be OK soon for using async/await in IL2CPP. It suppose that when Unity fixed issues with IL2CPP, it will stable enough for async.
So is there someone meet some unstable behavior with mono2x backend ? Or it is completely stable with mono2x now.

**Comment by ufcpp** [comment]
The MinimumAsyncBridge seems to be stable on Android with mono2x backend. In our projects, we have test plays on both iOS and Android and most problems are only on iOS.

**Comment by sharwell** [comment]
@ufcpp It's frustrating because the code works so well! :smile:
